src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Contracts/QueryRenameRequest.cs

```diff
@@ -0,0 +1,37 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.QueryExecution.Contracts
+{
+    /// <summary>
+    /// Parameters for the query cancellation request
+    /// </summary>
+    public class QueryRenameParams
+    {
+        public string NewOwnerUri { get; set; }
+        public string OriginalOwnerUri { get; set; }
+    }
+
+    /// <summary>
+    /// Parameters to return as the result of a query dispose request
+    /// </summary>
+    public class QueryRenameResult
+    {
+        /// <summary>
+        /// Any error messages that occurred during disposing the result set. Optional, can be set
+        /// to null if there were no errors.
```

Comment:
1. We should just throw an error if that happens I think - having it complete "successfully" but then have errors would be confusing I think
2. If we did do that you should generally also include a Successful property so it's easy to tell whether the request succeeded or not

Reply:
Error handling is handled in the handleRenameRequest task in QueryExecutionService, I don't see any other examples of QueryActionRequest that throw an error, if there is an example, can you please let me know?

Reply:
I'm saying that errors shouldn't be sent back in a result. You either call SendError or let the exception bubble up - both result in the caller on the ADS side getting an exception from the call with the error message from the exception.

Results should only be used for indicating success. And in this case since you don't really have any additional metadata I would say it would even be cleaner just to have the "response" be void instead of a whole object like this (that should be possible IIRC...)

Reply:
I'm following what is done in simpleExecuteRequest

Reply:
Alright I'll leave the renameresult empty, I'm following what is done in dispose which also returns an empty QueryDisposeResult object upon completion. If it is possible to remove the requestContext sendresult message from the code without any side effects, I'll do it, but there needs to be a QueryRenameResult object for the requestContext.